command/ingest.go

```diff
@@ -4,6 +4,7 @@ import (
 	"fmt"
 
 	httpclient "github.com/filecoin-project/storetheindex/api/v0/admin/client/http"
+	ingclient "github.com/filecoin-project/storetheindex/api/v0/ingest/client/http"
```

Comment:
why is this an ingest rather than an admin api action?
It seems like this is a listing that the operator of the storetheindex node would be able to do, and so would be exposed on the admin API. what's the distinction?

Reply:
I agree.  That was where it was for some time, probably never got moved when things were reorganized.  I will move it.  As far as I can tell it was not previously used anywhere so moving it will not affect anything.

Reply:
After giving it more thought, I think we want `ListProviders` and `GetProvider` to be part of the public ingest API.  That allows providers/publishers to confirm that their information is up-to-date at any indexer, and allows anyone to see if an indexer is indexing content for a particular provider.

Reply:
i think 1 level of abstraction meta from that:
when we say ‘the ingest api’ it would be great if we mean 1 specific thing:
the protocol for publishers to create and submit their chain of advertisements.

that we also have this http thing that’s used for interactive querying about the state of advertisements, but isn’t part of the ingest protocol overloads that term and is confusing.

This is not part of ‘the ingest api’ because it’s not available if i’m doing gossip sub + graphsync for transmission.

can we name it something else? ‘publisher status api’?

Reply:
There are three different services and each is available on a different port and potentially a different network interface:
- admin: 127.0.0.1:3002
- finder: 0.0.0.0:3000
- ingest: 0.0.0.0:3001

I was considering an API to be paired with each service. Finder and ingest were on different servers because of an anticipation that someone running an indexer for themselves may not want to expose the ingest service, but only the finder service, although these are combined in the libp2p service.

It is looking like I probably want to combine finder and ingest into one service, and then add the publisher status API to that service. 

- admin: 127.0.0.1:3002
- http: 0.0.0.0:3001

WDYT?

Reply:
* the HTTP should stay on :3000 as that's what infra just did a bunch of work to route differently
* I like the  http vs admin naming (or maybe 'public' vs 'admin' as the two services?) since these APIs are different from the ingest (and soon find) interfaces that we expect others to implement directly.

Reply:
I think "admin" and "indexer" make sense.  Indexer is served over http and libp2p.